rosplan_knowledge_base/src/KnowledgeBase.cpp

```diff
@@ -185,7 +185,7 @@ namespace KCL_rosplan {
     bool KnowledgeBase::updateKnowledgeArray(ros::ServiceEvent<rosplan_knowledge_msgs::KnowledgeUpdateServiceArray::Request, rosplan_knowledge_msgs::KnowledgeUpdateServiceArray::Response> &event) {
 
         rosplan_knowledge_msgs::KnowledgeUpdateServiceArray::Request req = event.getRequest();
```

Comment:
But the request does not need changed as well?

Reply:
It could be done to have both equal, but the request is an input parameter while the response is an output one. We only read the request, but modify the response as a return.